server/athenian/api/controllers/settings_controller.py

```diff
@@ -566,26 +544,39 @@ async def clean_repository_sets():
             model.acc_id == account,
             model.repository_full_name == full_name,
         ))))
+    clean_deployments_coro = _clean_logical_deployments(
+        [full_name, drop_logical_repo(full_name)], account, pdb,
+    )
+    await gather(*tasks, clean_deployments_coro, op="_delete_logical_repository/sql")
+
+
+async def _clean_logical_deployments(
+    repos: Sequence[str], account: int, pdb: DatabaseLike, op: str = None,
+) -> None:
+    """Delete all deployments info about `repos` from pdb."""
+    depl_models = (GitHubCommitDeployment, GitHubPullRequestDeployment, GitHubReleaseDeployment)
     affected_deployments = await pdb.fetch_all(union(*(
         select([distinct(model.deployment_name)]).where(and_(
```

Comment:
`delete ... returning` not yet in sqlalchemy with sqlite engine

Reply:
Yes. BTW can you please `log.info()` the `[r[0] for r in affected_deployments]`, it should simplify our life in the future.

Reply:
added